app/keeper/internal/route/store/health.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"crypto/fips140"
+	"encoding/json"
+	"net/http"
+
+	"github.com/spiffe/spike-sdk-go/log"
+	"github.com/spiffe/spike-sdk-go/security/mem"
+	"github.com/spiffe/spike-sdk-go/spiffe"
+	"github.com/spiffe/spike-sdk-go/spiffeid"
+
+	"github.com/spiffe/spike/app/keeper/internal/env"
+	"github.com/spiffe/spike/internal/net"
+
+	"github.com/spiffe/spike/internal/journal"
+)
+
+type HealthResponse struct {
+	Status        string `json:"status"`
+	MemLocked     bool   `json:"mem_locked"`
+	FIPS          bool   `json:"fips_enabled"`
+	ValidSPIFFEID bool   `json:"valid_spiffe_id"`
+}
+
+func RouteHealth(w http.ResponseWriter, r *http.Request, audit *journal.AuditEntry) error {
+	const fName = "routeHealth"
+	journal.AuditRequest(fName, r, audit, journal.AuditRead)
+
+	ctx := r.Context()
+	source, selfSPIFFEID, err := spiffe.Source(ctx, spiffe.EndpointSocket())
+	if err != nil {
+		log.Log().Error(fName, "message", "Cannot get SPIFFE source", "error", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return err
+	}
+	defer spiffe.CloseSource(source)
+
+	validSPIFFE := spiffeid.IsKeeper(env.TrustRoot(), selfSPIFFEID)
```

Comment:
If SPIFFE ID is not valid, do not return a response saying "hey my SPIFFEID is not valid"

An invalid SPIFFE ID means either there is a configuration problem, or this is a workload trying to impersonate a SPIKE Keeper.

Please check other parts of the codebase about that that control is done.

For example, SPIKE Nexus does not even boot up if its self-spiffe-id is invalid:
https://github.com/spiffe/spike/blob/2244b83a52d8867998b711bac3412c760fb74439/app/nexus/cmd/main.go#L58-L62